### PR TITLE
session,server: forbid stale read in local temporary tables

### LINKED ISSUE #25937: Forbid stale read in local temporary tables

### PR DESCRIPTION
**Pull request by sylzd**
### What problem does this PR solve?

Issue Number: close #25937 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by lcwangchao** [comment]
Why creating a new pr instead of #26976

File: planner/core/preprocess.go [hunk 1]
```diff
@@ -1594,10 +1594,23 @@ func (p *preprocessor) handleAsOfAndReadTS(node *ast.AsOfClause) {
 	}
 	if p.LastSnapshotTS != 0 {
 		dom := domain.GetDomain(p.ctx)
-		p.InfoSchema, p.err = dom.GetSnapshotInfoSchema(p.LastSnapshotTS)
-		if p.err != nil {
+		is, err := dom.GetSnapshotInfoSchema(p.LastSnapshotTS)
+		// if infoschema is empty, LastSnapshotTS init failed
+		if err != nil || is == nil {
+			p.err = err
```
Comment (lcwangchao) [inline planner/core/preprocess.go:1600]: If you think `GetSnapshotInfoSchema` will return a nil inforschema without any error. You should make a standalone if statement:

```
if err != nil {
  p.err = err
  return
}
if is == nil {
  p.err = errors.New("your error")
  return 
}
```

**Review by tiancaiamao — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**